examples/render/quad_render/main.rs

```diff
@@ -66,124 +81,46 @@ fn main() {
-        },
-    });
-    pipeline_desc.attributes.push(pso::AttributeDesc {
-        location: 1,
-        binding: 0,
-        element: pso::Element {
-            format: <Vec2<f32> as Formatted>::get_format(),
-            offset: 8
+    let sample_desc = context.mut_device().create_descriptors(1).pop().unwrap();
+    let pipe_init = pipe::Init {
+        sample: &sample_desc,
+        color: pso::ColorInfo {
+            mask: state::MASK_ALL,
+            color: Some(state::BlendChannel {
+                equation: state::Equation::Add,
+                source: state::Factor::ZeroPlus(state::BlendValue::SourceAlpha),
+                destination: state::Factor::OneMinus(state::BlendValue::SourceAlpha),
+            }),
+            alpha: Some(state::BlendChannel {
+                equation: state::Equation::Add,
+                source: state::Factor::One,
+                destination: state::Factor::One,
+            }),
         },
-    });
-
-    //
-    let pipelines = {
-        let shader_entries = pso::GraphicsShaderSet {
+        vertices: (),
+    };
+    let pipeline = context.mut_device().create_graphics_pipeline(
+        pso::GraphicsShaderSet {
             vertex: pso::EntryPoint { entry: "main", module: &vs_module },
             hull: None,
             domain: None,
             geometry: None,
             fragment: Some(pso::EntryPoint { entry: "main", module: &fs_module },),
-        };
-        let subpass = Subpass { index: 0, main_pass: &render_pass };
-        device.create_graphics_pipelines(&[
-            (shader_entries, &pipeline_layout, subpass, &pipeline_desc)
-        ])
-    };
-
-    println!("pipelines: {:?}", pipelines);
-
-    // Descriptors
-    let mut srv_pool = device.create_descriptor_pool(
-        1, // sets
-        &[pso::DescriptorRangeDesc { ty: pso::DescriptorType::SampledImage, count: 1 }],
-    );
-    let set0 = srv_pool.allocate_sets(&[&set0_layout]);
-
-    let mut sampler_pool = device.create_descriptor_pool(
-        1, // sets
-        &[pso::DescriptorRangeDesc { ty: pso::DescriptorType::Sampler, count: 1 }],
-    );
-    let set1 = sampler_pool.allocate_sets(&[&set1_layout]);
+        },
+        Primitive::TriangleList,
+        pso::Rasterizer::new_fill(),
+        pipe_init,
+    ).unwrap();
 
     // Framebuffer creation
-    let framebuffers = backbuffers.iter().map(|backbuffer| {
-        let frame_rtv = backbuffer.color.resource();
+    let frame_rtvs = backbuffers.iter().map(|backbuffer| {
+        context.mut_device()
+            .view_image_as_render_target(&backbuffer.color, (0..1, 0..1))
+            .unwrap()
+    }).collect::<Vec<_>>();
+    let framebuffers = frame_rtvs.iter().map(|rtv| {
         let extent = d::Extent { width: pixel_width as _, height: pixel_height as _, depth: 1 };
-        device.create_framebuffer(&render_pass, &[frame_rtv], &[], extent)
+        device.create_framebuffer(pipeline.render_pass(), &[rtv.resource()], &[], extent)
```

Comment:
I don't know if you have planned it but how difficult would you estimate to add custom renderpass support?

Reply:
I didn't think about it yet, I'm focusing on getting simple pipelines working for this PR.